include/solvers/petsc_dm_wrapper.h

```diff
@@ -31,13 +33,46 @@
 #include "libmesh/ignore_warnings.h"
 #include <petsc.h>
 #include "libmesh/restore_warnings.h"
+#include "libmesh/petsc_matrix.h"
+#include "libmesh/petsc_vector.h"
 
 namespace libMesh
 {
   // Forward declarations
   class System;
   class DofObject;
 
+  //! Struct to house data regarding where in the mesh hierarchy we are located
+  struct PetscDMContext
+  {
+    int n_dofs;
+    int mesh_dim;
+    DM * coarser_dm;
+    DM * finer_dm;
+    DM * global_dm;
+    libMesh::PetscMatrix<libMesh::Real > * K_interp_ptr;
```

Comment:
You are in namespace libMesh, so these `libMesh::` qualifications should be unnecessary? If they are just there for human code readability that's OK with me.